cookbooks/fb_ntp/recipes/default.rb

```diff
@@ -43,16 +47,14 @@
 
     # Resolve chosen timesources and allow them
     node['fb_ntp']['servers'].each do |host|
-      begin
-        ips = Resolv.getaddresses(host)
-        ips.each do |ip|
-          dash6 = ''
-          dash6 = '-6 ' if IPAddr.new(ip).ipv6?
-          acl_entries << "restrict #{dash6}#{ip} ##{host}"
-        end
-      rescue Resolv::ResolvError
-        Chef::Log.warn("fb_ntp: failed to resolve #{host}, skipping")
+      ips = Resolv.getaddresses(host)
+      ips.each do |ip|
+        dash6 = ''
+        dash6 = '-6 ' if IPAddr.new(ip).ipv6?
+        acl_entries << "restrict #{dash6}#{ip} ##{host}"
       end
+    rescue Resolv::ResolvError
+      Chef::Log.warn("fb_ntp: failed to resolve #{host}, skipping")
```

Comment:
This changes the error handling to skip single servers that don't resolve, to skipping all servers past the first one that fails. Why the change in semantics here?

Reply:
rubocop did this automatically. :( seems like a bug, you're right. :/

Reply:
Nope, wait, you're misunderstanding the code. The implicit rescue there doesn't stop the loop... it rescues. The code is identical:

```
[phil@rider ~]$ cat /tmp/test.rb 
#!/usr/bin/ruby

(0..9).each do |i|
  puts i
  raise if i == 5
rescue
  puts 'i am at 5'
end
[phil@rider ~]$ /tmp/test.rb
0
1
2
3
4
5
i am at 5
6
7
8
9
```

That code is correct.

Reply:
This sort of thing seems to be par for the course for Ruby. Alright, TIL then.